### PR TITLE
Use `AWS.Response` across test suite

### PR DESCRIPTION
**Pull request by omus**
Updates the tests from using the legacy response to the new `AWS.Response` in #384. I'm doing this in a separate PR to allow reviews to see the backwards compatibility mode (#384) separate from the use of `AWS.Response` in the tests (#458)

Depends on:
- #457
- #384

### CODE REVIEW

**Comment by omus** [comment]
bors try

**Comment by omus** [comment]
bors cancel

**Comment by omus** [comment]
bors try

**Comment by omus** [comment]
bors try

**Comment by omus** [comment]
bors try

**Comment by omus** [comment]
Using the `use_response_type` feature throughout the tests is working the high-level API calls. Now I'm working through the low-level tests

**Comment by omus** [comment]
bors try

**Comment by omus** [comment]
bors try

**Comment by omus** [comment]
bors try

**Comment by omus** [comment]
bors try

**Comment by omus** [comment]
bors try

**Comment by omus** [comment]
bors try

**Comment by omus** [comment]
bors try

**Comment by omus** [comment]
bors try

File: test/minio.jl [hunk 1]
```diff
@@ -42,6 +42,7 @@ try
     # Test retrieving an object into a stream target
     mktemp() do f, io
         S3.get_object("anewbucket", "myobject", Dict("response_stream" => io))
+        flush(io)
```
Comment (omus) [inline test/minio.jl:45]: Note I consider this a flaw with the tests and is not actually a breaking behaviour with using `AWS.Response`. Specifically, this test is writing to a temporary file and reading the contents before actually closing the `IOStream`

File: src/AWS.jl [hunk 2]
```diff
@@ -230,7 +230,7 @@ function (service::RestXMLService)(
     request_uri::String,
     args::AbstractDict{String,<:Any}=Dict{String,Any}();
     aws_config::AbstractAWSConfig=global_aws_config(),
-    feature_set::FeatureSet=FeatureSet(),
+    feature_set::FeatureSet,
```
Comment (omus) [inline src/AWS.jl:233]: Temporary change I was using to validate I found all internal AWS API calls

File: test/minio.jl (hunk 1 above)
Comment (omus) [reply test/minio.jl:45]: Was pulled out into a separate PR: #463

**Comment by omus** [comment]
bors try

**Comment by omus** [comment]
I'm going to wait on merging this PR about a week to ensure all of the bugs have been shaken out of #384.

**Comment by omus** [comment]
bors try

**Review by mattBrzezinski — APPROVED** [review]
LGTM, only question I have is do we want to only be testing the `use_response_type=true` functionality and not both?

**Comment by omus** [comment]
> LGTM, only question I have is do we want to only be testing the `use_response_type=true` functionality and not both?

A fair question. To me this is analogous to: should we test the deprecation and the new code. I'm not too concerned about this as we won't be supporting the deprecated code for too long (v2 should be out in a month or two?) but if we are concerned I'd probably go with adding tests for `legacy_response` only.

**Comment by omus** [comment]
@mattBrzezinski any feedback on the above?

**Review by mattBrzezinski — APPROVED** [review]
LGTM!

**Comment by omus** [comment]
Okay. I'll update and merge this.

**Comment by omus** [comment]
Going to wait on #522 as this improves on the tests from #516

**Pull request closed without merging**